sky/clouds/do.py

```diff
@@ -0,0 +1,275 @@
+""" Digital Ocean Cloud. """
+
+import json
+import typing
+from typing import Dict, Iterator, List, Optional, Tuple
+
+from sky import clouds
+from sky.clouds import service_catalog
+from sky.provision.do import utils
+from sky.utils import resources_utils
+
+if typing.TYPE_CHECKING:
+    from sky import resources as resources_lib
+
+_CREDENTIAL_FILE = 'config.yaml'
+
+
+@clouds.CLOUD_REGISTRY.register
+class DO(clouds.Cloud):
+    """Digital Ocean Cloud"""
+
+    _REPR = 'DO'
+    _CLOUD_UNSUPPORTED_FEATURES = {
+        clouds.CloudImplementationFeatures.CLONE_DISK_FROM_CLUSTER:
+            'Migrating '
+            f'disk is not supported in {_REPR}.',
+        clouds.CloudImplementationFeatures.SPOT_INSTANCE:
+            'Spot instances are'
+            f' not supported in {_REPR}.',
+        clouds.CloudImplementationFeatures.IMAGE_ID:
+            'Specifying image ID '
+            f'is not supported for {_REPR}.',
+        clouds.CloudImplementationFeatures.CUSTOM_DISK_TIER:
+            'Custom disk tiers'
+            f' is not supported in {_REPR}.',
+        clouds.CloudImplementationFeatures.DOCKER_IMAGE:
+            'Docker container images as runtime environments'
+            f' are not supported in {_REPR}. Try using in `run`',
```

Comment:
I included a startup script that installs docker, but ssh comes up before the startup script finishes running resulting in the docker environment setup failing. Docker does eventually get setup on the machine. If I for example, reattempt to launch on the same cluster after waiting for like 10 seconds, I can successfully use the docker environment, but the times i tested it never worked on freshly provisioned instances.

Reply:
Is it possible to use a base image with docker installed? Or, can we wait for the docker to finish setup for DO cloud? 

https://github.com/skypilot-org/skypilot/blob/e870839aeed16c118c0eb1f4889efc20006c27c4/sky/provision/docker_utils.py#L336-L343

Reply:
i added waiting for `docker` to `_check_docker_installed` but let me know if this should only be specific to `DO`

Reply:
My concern is that it will take too long to error out for an image without docker installation. Is it possible that we somehow detect if the docker installation is in progress, and immediately return if we found no installation?

Reply:
@cblmemo the most recent AI/ML images from digital ocean come with docker and are available now for CPU vms as well gpu vms. i'm going to revert the docker wait and just set this as the default image.